We might need another patch for x86_64-darwin.

<details><summary>error log on x86_64-darwin</summary>

```
=================================== FAILURES ===================================
_____________________________ test_shell_injection _____________________________
    def test_shell_injection():
        """Test injecting commands or shell args like `$0` into shell strings."""
        file = "./tests/test_files/injection.py"
        output = subprocess.check_output(["zxpy", file, "--", "abc"]).decode()
>       assert output == (
            "True\n"  # uname -p worked as a string
            "127\n"  # uname -p inside f-string got quoted
        )
E       AssertionError: assert 'False\n127\n' == 'True\n127\n'
E         - True
E         + False
E           127
tests/zxpy_test.py:170: AssertionError
```
</details>

--- a/pkgs/by-name/zx/zxpy/package.nix
+++ b/pkgs/by-name/zx/zxpy/package.nix
@@ -1,0 +1,53 @@
+{ lib
+, python3
+, fetchFromGitHub
+, fetchpatch
+, deterministic-uname
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "zxpy";
+  version = "1.6.3";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "tusharsadhwani";
+    repo = "zxpy";
+    rev = version;
+    hash = "sha256-/sOLSIqaAUkaAghPqe0Zoq7C8CSKAd61o8ivtjJFcJY=";
+  };
+
+  patches = [
+    # fix test caused by `uname -p` printing unknown
+    # https://github.com/tusharsadhwani/zxpy/pull/53
+    (fetchpatch {
+      name = "allow-unknown-processor-in-injection-test.patch";
+      url = "https://github.com/tusharsadhwani/zxpy/commit/95ad80caddbab82346f60ad80a601258fd1238c9.patch";
+      hash = "sha256-iXasOKjWuxNjjTpb0umNMNhbFgBjsu5LsOpTaXllATM=";
+    })
+  ];
+
+  nativeBuildInputs = [
+    python3.pkgs.setuptools
+    python3.pkgs.wheel
+  ];
+
+  nativeCheckInputs = [
+    deterministic-uname
+    python3.pkgs.pytestCheckHook
+  ];
+
+  preCheck = ''
+    export PATH=$out/bin:$PATH
+  '';
+
+  pythonImportsCheck = [ "zx" ];
+
+  meta = with lib; {
+    description = "Shell scripts made simple";
+    homepage = "https://github.com/tusharsadhwani/zxpy";
+    license = licenses.mit;
+    maintainers = with maintainers; [ figsoda ];
+    mainProgram = "zxpy";
+  };
+}

